### PR TITLE
feat: implement route to add a new education

### LINKED ISSUE #19: Implement API route to add new "Education"
On the `/resume/education` route, you should:

- [ ] Return the index position of the new Education in the Education list when a POST request is made

### PR DESCRIPTION
**Pull request by dannyl1u**
closes #19

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -118,9 +118,24 @@ def education():
...
-        return jsonify({})
+        if not request.is_json:
+            return jsonify({'error': 'Request must be JSON'}), 400
+
+        required_fields = ['course', 'school', 'start_date', 'end_date', 'grade', 'logo']
+        missing_fields = validate_fields(required_fields, request.json)
+
+        if missing_fields:
+            return jsonify({'error': f'Missing required fields: {", ".join(missing_fields)}'}), 400
+
```
Comment (Pradyuman7) [inline app.py:130]: nit: For start_date and end_date, are we sure that the frontend always sends the information in a specific format? Should we have some checks here to ensure that the dates are sent/added in correct format

Comment (dannyl1u) [reply app.py:130]: Hi @Pradyuman7, that's a good point but I think the backend should allow the frontend to send the date in any format (i.e. we accept any string), in order to account for different preferences (e.g. May 2024 vs 05/2024) or for current positions we would like to accept "Present" for `end_date` as well

Comment (Pradyuman7) [reply app.py:130]: Hmm, this can indeed be an approach. Could you add this comment link to the frontend issue so this doesn't get ignored?

**Review by Pradyuman7 — APPROVED** [review]
LGTM 👍 
I'll merge once the conflicts are resolved

File: app.py (hunk 1 above)
Comment (dannyl1u) [reply app.py:130]: Done!

**Comment by dannyl1u** [comment]
> LGTM 👍 I'll merge once the conflicts are resolved

@Pradyuman7 done, thank you!

**Comment by Pradyuman7** [comment]
Haha, conflicts again, if you can resolve, I'll merge them

**Pull request merged**